frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditActions/EditActions.test.tsx

```diff
@@ -0,0 +1,138 @@
+  signingActions,
+} from '../../../../../test/mocks/bpmnDetailsMock';
+import {
+  createMock,
+  mockModelerRef,
+  updateModdlePropertiesMock,
+} from '../../../../../test/mocks/bpmnModelerMock';
+import { useBpmnContext } from '../../../../contexts/BpmnContext';
+import type { BpmnDetails } from '../../../../types/BpmnDetails';
+import type { BpmnTaskType } from '../../../../types/BpmnTaskType';
+import userEvent from '@testing-library/user-event';
+import { ObjectUtils } from '@studio/pure-functions';
+
+const actionsForTaskTypes = { confirmation: confirmationActions, signing: signingActions }; // add payment: paymentActions
+
+const setBpmnDetailsMock = jest.fn();
+jest.mock('../../../../contexts/BpmnContext', () => ({
+  useBpmnContext: jest.fn(() => ({
+    modelerRef: mockModelerRef,
+    setBpmnDetails: setBpmnDetailsMock,
+    bpmnDetails: mockBpmnDetails,
+  })),
+}));
+
+jest.mock('bpmn-js/lib/features/modeling/BpmnFactory', () => ({
+  BpmnFactory: jest.fn(() => ({
+    create: jest.fn(),
+  })),
+}));
+
+const overrideBpmnDetailsMock = (bpmnDetailsToOverride: BpmnDetails) => {
+  (useBpmnContext as jest.Mock).mockReturnValue({
+    modelerRef: mockModelerRef,
+    setBpmnDetails: setBpmnDetailsMock,
+    bpmnDetails: bpmnDetailsToOverride,
+  });
+};
+
+describe('EditActions', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render only "add new action" button when task have no actions', () => {
+    renderEditActions({
+      ...mockBpmnDetails,
+      element: { businessObject: { extensionElements: { values: [{}] } } },
+    });
+    const addNewActionButton = screen.getAllByRole('button', {
+      name: textMock('process_editor.configuration_panel_actions_add_new'),
+    });
+    expect(addNewActionButton).toHaveLength(1);
+  });
+
+  it('should render existing actions when task have predefined actions', () => {
+    const predefinedAction =
+      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[0].action;
+    const customServerAction =
+      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[1].action;
+    const customProcessAction =
+      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[2].action;
+    renderEditActions();
+    screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_actions_action_label', {
+        actionIndex: 1,
+        actionName: predefinedAction,
+      }),
+    });
+    screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_actions_action_label', {
+        actionIndex: 2,
+        actionName: customServerAction,
+      }),
+    });
+    screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_actions_action_label', {
+        actionIndex: 3,
+        actionName: customProcessAction,
+      }),
+    });
+  });
+
+  it.each(['confirmation', 'signing'])(
+    // add payment
```

Comment:
Same "issue" as above. Dont know when payment should be released so figured I should wait, but also as mentioned above, since adding a payment task to the diagram is featureflagged no one should get the config panel for payment task anyway so we can add it 🥳